krun/scheduler.py

```diff
@@ -350,131 +346,134 @@ def run(self):
+                         "ETA for Current Session Available",
+                         msg, bypass_limiter=True)
+
+        info("{:<25s}: {} ({} from now)".format(
+            "Estimated completion (whole session)", tfmt.finish_str,
+            tfmt.delta_str))
+
+        info("%d executions left in scheduler queue" % self.manifest.num_execs_left)
+
+        if self.manifest.num_execs_left > 0 and \
+                self.manifest.eta_avail_idx > self.manifest.next_exec_idx:
+            info("Executions until ETA known: %s" %
+                 (self.manifest.eta_avail_idx -
+                  self.manifest.next_exec_idx))
+
+        if self.platform.check_dmesg_for_changes():
+            self.results.error_flag = True
+
+        if self.manifest.num_execs_left > 0:
+            # print info about the next job
+            benchmark, vm_name, variant = \
+                self.manifest.next_exec_key.split(":")
+            info("Next execution is '%s(%d)' (%s variant) under '%s'" %
+                 (benchmark, self.config.BENCHMARKS[benchmark], variant, vm_name))
+
+            tfmt = self.get_exec_estimate_time_formatter(job.key)
+            info("{:<35s}: {} ({} from now)".format(
+                "Estimated completion (next execution)",
+                tfmt.finish_str,
                 tfmt.delta_str))
 
-            info("%d executions left in scheduler queue" % self.manifest.num_execs_left)
-
-            if self.manifest.num_execs_left > 0 and \
-                    self.manifest.eta_avail_idx > self.manifest.next_exec_idx:
-                info("Executions until ETA known: %s" %
-                     (self.manifest.eta_avail_idx -
-                      self.manifest.next_exec_idx))
-
-            if self.platform.check_dmesg_for_changes():
-                self.results.error_flag = True
-
-            if self.manifest.num_execs_left > 0:
-                # print info about the next job
-                benchmark, vm_name, variant = \
-                    self.manifest.next_exec_key.split(":")
-                info("Next execution is '%s(%d)' (%s variant) under '%s'" %
-                     (benchmark, self.config.BENCHMARKS[benchmark], variant, vm_name))
-
-                tfmt = self.get_exec_estimate_time_formatter(job.key)
-                info("{:<35s}: {} ({} from now)".format(
-                    "Estimated completion (next execution)",
-                    tfmt.finish_str,
-                    tfmt.delta_str))
-            elif self.manifest.num_execs_left == 0:
-                break  # done
-            else:
-                assert False
+            info("Reboot in preparation for next execution")
+            self._reboot(self.manifest.total_num_execs)
 
-            if self.reboot and self.manifest.num_execs_left > 0:
-                info("Reboot in preparation for next execution")
-                self._reboot(self.manifest.total_num_execs)
+        elif self.manifest.num_execs_left < 0:
+            assert False
```

Comment:
This is is not a good `assert False` -- it should be somewhere else (e.g. before the `if`).